service/frontend/redirection_interceptor.go

```diff
@@ -177,8 +177,9 @@ func (i *RedirectionInterceptor) Intercept(
 		return i.handleRedirectAPIInvocation(ctx, req, info, handler, methodName, respAllocFn, namespaceName)
 	}
 
-	// this should not happen, log warn and move on
-	i.logger.Warn(fmt.Sprintf("RedirectionInterceptor encountered unknown API: %v", methodName))
+	// this should not happen for frontend APIs but ok for admin and other future handlers
```

Comment:
You can get the service name from info.FullMethod and log at Warn level for WorkflowService, and Debug level for AdminService or others. That might be more useful to catch bugs

Reply:
https://github.com/temporalio/temporal/blob/master/service/frontend/redirection_interceptor_test.go#L183

all workflow service APIs are validated